### PR TITLE
[5.1] Proposal for handling multiple database slave

### PR DESCRIPTION
**Pull request by premsaurav**

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Please apply the diff from StyleCI.

**Comment by premsaurav** [comment]
Done. Thanks

File: src/Illuminate/Database/Connectors/ConnectionFactory.php [hunk 1]
```diff
@@ -98,6 +98,13 @@ protected function getReadConfig(array $config)
     {
         $readConfig = $this->getReadWriteConfig($config, 'read');
 
+        // Check if we have multiple slave hosts
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Connectors/ConnectionFactory.php]: don't think the config adds any value

**Comment by premsaurav** [comment]
Sorry, didn't get that. Are you talking about this modification not adding any value ?

Motivation: 1 master - X slaves. At run time, pick a slave at random for all read queries. Very common scenario. Can do the same is config/databases.php , but not sure how php artisan config:cache will behave then.

File: src/Illuminate/Database/Connectors/ConnectionFactory.php (hunk 1 above)
Comment (premsaurav) [inline src/Illuminate/Database/Connectors/ConnectionFactory.php]: Sorry, didn't get that. Are you talking about this modification not adding any value ?

Motivation: 1 master - X slaves. At run time, pick a slave at random for all read queries. Very common scenario. Can do the same is config/databases.php , but not sure how php artisan config:cache will behave then.

Comment (GrahamCampbell) [inline src/Illuminate/Database/Connectors/ConnectionFactory.php]: meant to say "comment" not "config" - sorry

Comment (premsaurav) [inline src/Illuminate/Database/Connectors/ConnectionFactory.php]: fixed. Thanks

**Comment by GrahamCampbell** [comment]
Please squash to one commit. ;)

**Comment by premsaurav** [comment]
Having trouble :( will close this and open new pull :(

**Pull request closed without merging**